gmakefile.test

```diff
@@ -83,13 +88,8 @@ ifdef TIMEOUT
   TESTFLAGS += -t $(TIMEOUT)  # Override the default timeout
 endif
 
-PETSCCONFIGDIR ?= $(PETSC_DIR)/config
-ifeq ($(wildcard $(PETSCCONFIGDIR)/gmakegentest.py),)
-  PETSCCONFIGDIR := $(PETSC_DIR)/share/petsc/examples/config
-endif
```

Comment:
aha ok, I though it was some old version legacy - will revert

Reply:
But shouldn't we just check for empty PETSC_ARCH?

Reply:
Empty PETSC_ARCH is a different use case from prefix install. With empty PETSC_ARCH gmakegentest is still in $(PETSC_DIR)/config.

Reply:
I wonder what does empty PETSC_ARCH with no prefix install even mean?

Reply:
There is no PETSC_ARCH with prefix installs. It is primarily used in packages and modules. See. https://www.mcs.anl.gov/petsc/documentation/installation.html#root

Reply:
My point is what is the difference between "empty PETSC_ARCH" and "prefix install" case. As you just wrote, there is no PETSC_ARCH with prefix installs. So why not consider "empty PETSC_ARCH" implying the "prefix install" case.

Is it possible to have non-prefix PERMON on prefix PETSc and vice-versa?

Reply:
Empty PETSC_ARCH does not imply prefix install. Prefix install implies empty PETSC_ARCH. If I remember correctly you can have installs with:
PETSC_DIR+PETSC_ARCH - lib is in $PETSC_DIR/$PETSC_ARCH/lib/
PETSC_DIR+ empty PETSC_ARCH - lib is in $PETSC_DIR/lib/?
prefix - lib is in $prefix/lib ? and lot of makefiles and scripts are somewhere under $prefix and not in the same location as in the standard installs

PERMON does not support prefix - there is no configure.

Reply:
> PETSC_DIR+ empty PETSC_ARCH - lib is in $PETSC_DIR/lib/?
```
$ ./configure PETSC_ARCH=""
===============================================================================
             Configuring PETSc to compile on your system                       
===============================================================================
TESTING: configureArchitecture from PETSc.options.arch(config/PETSc/options/arch.py:43)                                                                                                                             *******************************************************************************
         UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
-------------------------------------------------------------------------------
PETSC_ARCH cannot be empty string. Use a valid string or do not set one. Currently set with option PETSC_ARCH=
*******************************************************************************
```
So your option 2 does not exist, and I think we are safe to assume that  `PETSC_ARCH=""` implies PETSC being prefix-installed (with installation finished) with `PETSC_DIR=$prefix`.

> PERMON does not support prefix - there is no configure.

Yes, I realized that `prefix-installed PETSc => PETSC_ARCH="" => PERMON's lib directory is $PERMON_DIR/lib`. I think it's ok for now.

Reply:
It is not OK. By empty I mean it is not set. Setting `PETSC_ARCH=""` does not make it empty, you need to `unset PETSC_ARCH`

I now remembered, that I spend quite a while making PERMON work with prefix installs of PETSc https://github.com/permon/permon/commit/bb9dcb1525db64a4acfbb6ce3c40b2785de3703c

In any case, this should be tested with all the 3 configs I mentioned above.

Reply:
> It is not OK.

I meant the current PERMON behavior that it does not really support prefix install and installs into its root directory instead.

> By empty I mean it is not set. Setting PETSC_ARCH="" does not make it empty, you need to unset PETSC_ARCH

OK, with PETSC_ARCH being unset, configure just assumes some default arch name. In my case it's `arch-linux2-c-debug`. It does not install everything into PETSC_DIR directly! Anyway, for us it's not a separate option because we then need to set this as `PETSC_ARCH`. So for us only your options 1 and 3 are relevant.

Reply:
> OK, with PETSC_ARCH being unset, configure just assumes some default arch name. In my case it's `arch-linux2-c-debug`. It does not install everything into PETSC_DIR directly! Anyway, for us it's not a separate option because we then need to set this as `PETSC_ARCH`. So for us only your options 1 and 3 are relevant.

OK.

Did you test the removal of the code above with a prefix PETSc install?

Reply:
I now pushed 2f2b027 with the PETSC_ARCH-based decision and it's working for both variants (prefix/no prefix).